xmake/plugins/lua/xmake.lua

```diff
@@ -56,8 +61,33 @@ task("lua")
                 import("scripts." .. name).main(unpack(option.get("arguments") or {}))
             end
         else
+            local replhistory
+            if enable_readline then
+                -- clear history
+                readline.clear_history()
+
+                -- load history
+                replhistory = history.load("replhistory") or {}
+                for _, ln in ipairs(replhistory) do
+                    readline.add_history(ln)
+                end
+            end
+
             -- enter interactive mode
             sandbox.interactive()
+
```

Comment:
You can use `require("lib/readline")` and `require("base/history")` to handle history in [sandbox_core_sandbox.interactive()](https://github.com/tboox/xmake/blob/master/xmake/core/sandbox/modules/import/core/sandbox/sandbox.lua) and simplify the implementation in `lua action` as far as possible.